.travis.yml

```diff
@@ -1,24 +1,43 @@
 language: php
 
-php:
-  - 5.6
-  - 5.5
-  - 5.4
-  - 7
-  - hhvm
-  - hhvm-nightly
-
 matrix:
-  fast_finish: true
-  allow_failures:
-    - php: hhvm-nightly
-    - php: 7
+    include:
+        - php: 5.4
+        - php: 5.5
+        - php: 5.6
+        - php: 7.0
+        - php: 7.1
+        - php: nightly
+        - php: hhvm-3.6
+          sudo: required
+          dist: trusty
+          group: edge
+        - php: hhvm-3.9
+          sudo: required
+          dist: trusty
+          group: edge
+        - php: hhvm-3.12
+          sudo: required
+          dist: trusty
+          group: edge
+        - php: hhvm-3.15
+          sudo: required
+          dist: trusty
+          group: edge
+        - php: hhvm-nightly
+          sudo: required
+          dist: trusty
+          group: edge
+    fast_finish: true
+    allow_failures:
+        - php: nightly
+        - php: hhvm-nightly
 
 install:
   - composer install --prefer-source
 
 script:
-  - CODE_COVERAGE=clover bin/peridot specs/
+  - CODE_COVERAGE=clover make coverage
 
-after_script:
-  - php vendor/bin/coveralls
+after_success:
+  - bash <(curl -s https://codecov.io/bash)
```

Comment:
Looks good, but there's some important little performance hacks missing here:

```yaml
cache:
    directories:
        - $HOME/.composer
```

^ This will allow us to use Composer's cache across builds. It can have a pretty big impact on build time.

```yaml
sudo: false
```

^ This cryptic line tells Travis that we want to run on their new container infrastructure, which boots up much faster. It's apparently only necessary for repos that were created before this option became available. Newer repos get it by default. But I'm guessing that Peridot is old enough that it needs it.